File: TestAutoHelperJava/src/org/usfirst/frc/team195/robot/Robot.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

package org.usfirst.frc.team195.robot;

import edu.wpi.first.wpilibj.SampleRobot;
import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.Spark;
import edu.wpi.first.wpilibj.Timer;
import edu.wpi.first.wpilibj.drive.DifferentialDrive;
import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

/**
 * This is a demo program showing the use of the RobotDrive class. The
 * SampleRobot class is the base of a robot application that will automatically
 * call your Autonomous and OperatorControl methods at the right time as
 * controlled by the switches on the driver station or the field controls.
 *
 * <p>The VM is configured to automatically run this class, and to call the
 * functions corresponding to each mode, as described in the SampleRobot
 * documentation. If you change the name of this class or the package after
 * creating this project, you must also update the build.properties file in the
 * project.
 *
 * <p>WARNING: While it may look like a good choice to use for your code if
 * you're inexperienced, don't. Unless you know what you are doing, complex code
 * will be much more difficult under this system. Use IterativeRobot or
 * Command-Based instead if you're new.
 */
public class Robot extends SampleRobot {
	private Spark s0 = new Spark(0);
	private Spark s1 = new Spark(1);
	private DifferentialDrive m_robotDrive = new DifferentialDrive(s0, s1);
	private Joystick m_stick = new Joystick(0);
	private SendableChooser<String> m_chooser = new SendableChooser<>();

	private CKAutoBuilder<Spark> ckAuto;
	
	public Robot() {
		m_robotDrive.setExpiration(0.1);
		
		try {
			ckAuto = new CKAutoBuilder<Spark>(s0, s1, this);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	@Override
	public void robotInit() {

	}


	@Override
	public void autonomous() {
		ckAuto.addAutoStep(0, 0, 3000);
		ckAuto.addAutoStep(1, 0, 5000);
		ckAuto.addAutoStep(1, 0.25, 2000);
		ckAuto.addAutoStep(0, 0, 200);
		ckAuto.start();
		while(!isOperatorControl() && isEnabled()) {
			try {
				Thread.sleep(100);
			} catch (InterruptedException e) {
				;
			}
		}
	}

	@Override
	public void operatorControl() {
		m_robotDrive.setSafetyEnabled(true);
		while (isOperatorControl() && isEnabled()) {
			// Drive arcade style
			m_robotDrive.arcadeDrive(-m_stick.getY(), m_stick.getX());

			// The motors will be updated every 5ms
			Timer.delay(0.005);
		}
	}

	@Override
	public void test() {
	}
}

File: TestAutoHelper/src/Robot.cpp
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

#include <iostream>
#include <string>

#include "WPILib.h"
#include "CKAutoBuilder.h"
#include <Drive/DifferentialDrive.h>
#include <Joystick.h>
#include <SampleRobot.h>
#include <SmartDashboard/SendableChooser.h>
#include <SmartDashboard/SmartDashboard.h>
#include <Spark.h>
#include <Timer.h>

/**
 * This is a demo program showing the use of the DifferentialDrive class.
 * The SampleRobot class is the base of a robot application that will
 * automatically call your Autonomous and OperatorControl methods at the right
 * time as controlled by the switches on the driver station or the field
 * controls.
 *
 * WARNING: While it may look like a good choice to use for your code if you're
 * inexperienced, don't. Unless you know what you are doing, complex code will
 * be much more difficult under this system. Use IterativeRobot or Command-Based
 * instead if you're new.
 */
class Robot : public frc::SampleRobot {
public:
	Robot() {
		ckAuto = new CKAutoBuilder<Spark>(&m_leftMotor, &m_rightMotor, this);
	}

	void RobotInit() {

	}

	/*
	 * This autonomous (along with the chooser code above) shows how to
	 * select between different autonomous modes using the dashboard. The
	 * sendable chooser code works with the Java SmartDashboard. If you
	 * prefer the LabVIEW Dashboard, remove all of the chooser code and
	 * uncomment the GetString line to get the auto name from the text box
	 * below the Gyro.
	 *
	 * You can add additional auto modes by adding additional comparisons to
	 * the if-else structure below with additional strings. If using the
	 * SendableChooser make sure to add them to the chooser code above as
	 * well.
	 */
	void Autonomous() {
		ckAuto->addAutoStep(0, 0, 3000);
		ckAuto->addAutoStep(1, 0, 5000);
		ckAuto->addAutoStep(1, 0.25, 2000);
		ckAuto->addAutoStep(0, 0, 200);
		ckAuto->start();
		while(!IsOperatorControl() && IsEnabled()) {this_thread::sleep_for(chrono::milliseconds(100));}
	}

	/*
	 * Runs the motors with arcade steering.
	 */
	void OperatorControl() override {
		m_robotDrive.SetSafetyEnabled(true);
		while (IsOperatorControl() && IsEnabled()) {
			// Drive with arcade style (use right stick)
			m_robotDrive.ArcadeDrive(
					-m_stick.GetY(), m_stick.GetX());

			// The motors will be updated every 5ms
			frc::Wait(0.005);
		}
	}

	/*
	 * Runs during test mode
	 */
	void Test() override {}

private:
	// Robot drive system
	frc::Spark m_leftMotor{0};
	frc::Spark m_rightMotor{1};
	frc::DifferentialDrive m_robotDrive{m_leftMotor, m_rightMotor};

	frc::Joystick m_stick{0};

	CKAutoBuilder<Spark> *ckAuto;
};

START_ROBOT_CLASS(Robot)
